cmake/defaults/gccclangshareddefaults.cmake

```diff
@@ -42,8 +42,14 @@ endif()
 _disable_warning("deprecated")
 _disable_warning("deprecated-declarations")
 
-# Suppress unused typedef warnings emanating from boost.
-_disable_warning("unused-local-typedefs")
+# Suppress unused typedef warnings eminating from boost.
```

Comment:
I fixed the typo.
Perhaps I'm not understanding it right, but the other request seems a more confusing logic.

Instead of `if (!clang || clang > 3.5)`, you'd like  `if (!clang || !(clang < 3.6))`?